### PR TITLE
Document includeBuild for plugin builds

### LINKED ISSUE #16314: Document new includeBuild for plugin builds
The `pluginManagement { includeBuild("...") }` construct is mentioned in release notes but not yet documented in the user manual. Since this is incubating for now, we do not need to update all samples/snippets that should eventually use it right now. But we should at least have a short section about how to use this for settings plugins.

Also:
- This section needs an update and link to the new section on setting plugins:
   -  https://docs.gradle.org/current/userguide/structuring_software_products_details.html#settings_convention_plugins
- The large project sample should already use the new mechanism (it also uses other incubating features):
   - https://docs.gradle.org/current/samples/sample_structuring_software_projects.html

### PR DESCRIPTION
**Pull request by jjohannes**
Fixes #16314

New section:
https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/41913125:id/distributions/gradle-7.0-docs.zip!/gradle-7.0-20210308100502%2B0000/docs/userguide/composite_builds.html#included_plugin_builds

### CODE REVIEW

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/authoring-builds/software-product/composite_builds.adoc [hunk 1]
```diff
@@ -80,6 +78,23 @@ include::sample[dir="samples/build-organization/composite-builds/basic/kotlin",f
...
+Using this mechanism, the included build may also contribute a settings plugin that can be applied in the settings file itself.
+
+.Including a plugin build
+====
+include::sample[dir="snippets/developingPlugins/testingPlugins/groovy/include-plugin-build",files="settings.gradle[tags=include-build]"]
+include::sample[dir="snippets/developingPlugins/testingPlugins/kotlin/include-plugin-build",files="settings.gradle.kts[tags=include-build]"]
+====
+
+NOTE: Including plugin builds via the plugin management block is an incubating feature.
+You may also use the normal `includeBuild` mechanism to include plugin builds.
```
Comment (rieske) [inline subprojects/docs/src/docs/userguide/authoring-builds/software-product/composite_builds.adoc:95]: Maybe instead of saying "normal `includeBuild` mechanism", rephrase it to something like "stable `includeBuild` mechanism outside of `pluginManagement` block".

**Review by rieske — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/authoring-builds/software-product/composite_builds.adoc (hunk 1 above)
Comment (jjohannes) [reply subprojects/docs/src/docs/userguide/authoring-builds/software-product/composite_builds.adoc]: Good point. I rephrased this.
